squashfs/index.html

```diff
@@ -1042,6 +1233,150 @@ <h1>UID/GID Table</h1>
+                    <h4>Xattr Value Entry</h4>
+                    <table class="table table-sm">
+                        <thead>
+                            <tr>
+                                <th>Name</th>
+                                <th>Type</th>
+                                <th>Description</th>
+                            </tr>
+                        </thead>
+                        <tbody>
+                            <tr>
+                                <td>size</td>
+                                <td>u32</td>
+                                <td>The size of the value string</td>
+                            </tr>
+                            <tr>
+                                <td>value</td>
+                                <td>u8[]</td>
+                                <td>The raw value assigned to the key</td>
+                            </tr>
+                        </tbody>
+                    </table>
+                    <p>
+                        To actually address a block of key value pairs associated with an inode, a lookup table is used that specifies the start and size of a block of key value pairs.
+                    </p>
+                    <p>
+                        All an inode needs to store is a 32 bit index into this table. If two inodes have the identical xattrs (e.g. they have the same SELinux labels and no other attributes), the key/value block is only written once, there is only one lookup table entry and both inodes have the same index.
+                    </p>
+                    <h4>Xattr Lookup Table</h4>
+                    <table class="table table-sm">
+                        <thead>
+                            <tr>
+                                <th>Name</th>
+                                <th>Type</th>
+                                <th>Description</th>
+                            </tr>
+                        </thead>
+                        <tbody>
+                            <tr>
+                                <td>xattr_ref</td>
+                                <td>u64</td>
+                                <td>A reference to the start of the key value block. Similar to <a href="#inode-references">inode references</a>, the bits 16 to 48 hold the metadata block index and the lower 16 bit hold an offset into the uncompressed block</td>
+                            </tr>
+                            <tr>
+                                <td>count</td>
+                                <td>u32</td>
+                                <td>The number of key value pairs</td>
+                            </tr>
+                            <tr>
+                                <td>size</td>
+                                <td>u32</td>
+                                <td>The exact, uncompressed size in bytes of the entire block of key value pairs, counting only what has been written to disk and including the key/value entry structures</td>
```

Comment:
This includes both inline and out of line values?

Reply:
I'm not entirely sure yet how the out-of-line storage works and will have to investigate further.